### PR TITLE
xdg-desktop-portal: Fix `installedTests`

### PR DESCRIPTION
**Pull request by Artturin**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/xdg-desktop-portal/default.nix [hunk 1]
```diff
@@ -142,6 +144,22 @@ stdenv.mkDerivation (finalAttrs: {
...
 
+  postFixup = let
+    documentFuse = "${placeholder "installedTests"}/libexec/installed-tests/xdg-desktop-portal/test-document-fuse.py";
+    testPortals = "${placeholder "installedTests"}/libexec/installed-tests/xdg-desktop-portal/test-portals";
+
+  in ''
+    if [ -x '${documentFuse}' ] ; then
+      wrapGApp '${documentFuse}'
+      wrapGApp '${testPortals}'
+      cp -r tests/share/applications $installedTests/share
```
Comment (jtojnar) [inline pkgs/development/libraries/xdg-desktop-portal/default.nix:155]: Why does not upstream install this?

Comment (Artturin) [reply pkgs/development/libraries/xdg-desktop-portal/default.nix:155]: https://github.com/flatpak/xdg-desktop-portal/pull/1444

Comment (Artturin) [reply pkgs/development/libraries/xdg-desktop-portal/default.nix]: Hmm now it simply doesn't work and doesn't detect the files even though the share is listed and contains thefiles

```
vm-test-run-xdg-desktop-portal> # xdg-desktop-portal-DEBUG: Setting XDG_DATA_DIRS to /nix/store/18anzwnlkpl348yg1j94hikywc39kaah-xdg-desktop-portal-1.18.4/share:/nix/store/99kdsvjp77yv0yxs9wzpn8x227j7d1zc-gsettings-desktop-schemas-46.0/share/gsettings-sche
mas/gsettings-desktop-schemas-46.0:/nix/store/nzll3s1gzpmqiij4vyanpnpybryx8i0f-gtk+3-3.24.43/share/gsettings-schemas/gtk+3-3.24.43:/nix/store/fm8lwmakyhk770r9vc6db4bf68mcm88g-pipewire-1.2.3/share/gsettings-schemas/pipewire-1.2.3:/nix/store/j6s741bjd5wl3pa5
lwq5j7lg0s83z0fg-xdg-desktop-portal-1.18.4-installedTests/share/installed-tests/xdg-desktop-portal/share:/root/.nix-profile/share:/nix/profile/share:/root/.local/state/nix/profile/share:/etc/profiles/per-user/root/share:/nix/var/nix/profiles/default/share:
/run/current-system/sw/share:/usr/local/share/:/usr/share/
```

Comment (Artturin) [reply pkgs/development/libraries/xdg-desktop-portal/default.nix]: Had to install to `.../share/applications` now

File: nixos/tests/installed-tests/xdg-desktop-portal.nix [hunk 2]
```diff
@@ -3,7 +3,25 @@
...
-  # https://github.com/flatpak/xdg-desktop-portal/pull/428
-  meta.broken = true;
+  # Red herring
+  # Failed to load RealtimeKit property: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.RealtimeKit1 was not provided by any .service files
+  # Maybe a red herring, enabling PipeWire doesn't fix the location test.
+  # Failed connect to PipeWire: Couldn't connect to PipeWire
+  testConfig = {
+    xdg.portal = {
+      enable = true;
+      extraPortals = [ pkgs.xdg-desktop-portal-gtk ];
```
[the remark this answers is not in the document]
Comment (Artturin) [reply nixos/tests/installed-tests/xdg-desktop-portal.nix]: Indeed it's not necessary, I added it when trying to get it to work and didn't think to remove it and test if it still works.

File: pkgs/development/libraries/xdg-desktop-portal/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/development/libraries/xdg-desktop-portal/default.nix]: It's been merged

**Pull request merged**